lib/question.js

```diff
@@ -349,10 +349,19 @@ module.exports = {
             },
             (callback) => { // if we have a workspace, get the files
                 if (workspace_id == null) return callback(null);
-                util.callbackify(workspace.getGradedFiles)(workspace_id, (err, result) => {
+                sqldb.queryZeroOrOneRow(sql.select_workspace_graded_files, {workspace_id}, (err, result) => {
                     if (ERR(err, callback)) return;
-                    zipPath = result;
-                    callback(null);
+                    let workspace_graded_files = [];
+                    if (result.rowCount > 0) {
+                        workspace_graded_files = result.rows[0].workspace_graded_files;
+                    }
+                    submission.submitted_answer['_required_file_names'] = workspace_graded_files;
```

Comment:
I think we should:
1. Also set `submission.submitted_answer['_workspace_required_file_names']` to the same value, and use that in the `parse()` function. This will let us distinguish files required by the workspace versus files required by other elements.
2. Don't overwrite `_required_file_names` if it already exists, but instead append to it.

Reply:
Done. I also duped `_workspace_required_file_names` into `_required_file_names`. Not sure if you still wanted this (I guess that was pre-edit?).

Reply:
It doesn't really matter either way at this point (we never use it, but all file creation elements consistently make it). I'm fine to have it there.